SDK_versioned_docs/version-3.0.0/guides/04-execute-a-swap.md

```diff
@@ -0,0 +1,249 @@
+---
+id: execute-a-swap
+title: Executing A Swap
+---
+
+This guide extends the previous [creating a pool](03-creating-a-pool.md) and [getting started](./using-ethers) guides by using the pool object to quote an estimated amount out for a trade, then creates a trade that can be submitted to the EVM.
+
+## Creating A Quoter Contract Object
+
+In order to retrieve a quote, we need to create a [Contract](https://docs.ethers.io/v5/api/contract/contract/) object using ethers.js
+
+The quoter is a smart contract that retrieves estimated output or input amounts for a given swap type. For our example, we will create an object that models the [quoter interface](https://github.com/Uniswap/uniswap-v3-periphery-optimism/blob/69e095f7c28a7e7fdaca94b5eaa644a8a25f86cc/contracts/interfaces/IQuoter.sol), so we may call it and return a swap quote.
+
+Create the quoter contract object by importing the [ABI](https://docs.soliditylang.org/en/v0.7.0/abi-spec.html) from the [uniswap-v3-periphery](https://www.npmjs.com/package/@uniswap/v3-periphery) npm package, assign your [ethereum endpoint provider](https://ethereum.org/en/developers/docs/nodes-and-clients/nodes-as-a-service/), in this case, Infura, and provide the [deployment address of the quoter contract](https://github.com/Uniswap/uniswap-v3-periphery/blob/main/deploys.md).
```

Comment:
I totally followed this, but to simplify for newcomers I suggest breaking this up into three distinct sentences. Especially for people with english as a second language, complex sentence structure can make it difficult to understand

Reply:
I wonder if we should even break the code example into three separate code blocks, with a sentence of text in between each?  I can easily infer the mapping of sentence 1 -> line of code 1, but newcomers may not see it right way